src/Xamarin.Android.Build.Tasks/Tasks/AppendCustomMetadataToItemGroup.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tasks {
+	public class AppendCustomMetadataToItemGroup : Task {
+		[Required]
+		public ITaskItem[] Inputs { get; set; }
+
+		[Required]
+		public ITaskItem[] MetaDataItems { get; set; }
+
+		[Output]
+		public ITaskItem[] Output { get; set; }
+
+		public override bool Execute ()
+		{
+			var output = new List<ITaskItem> ();
+
+			foreach (var item in Inputs) {
+				var fn = Path.GetFileNameWithoutExtension (item.ItemSpec);
+				output.Add (item);
+				foreach (var metaData in MetaDataItems) {
```

Comment:
This is an O(N**2) algorithm, iterating `Inputs.Length*MetaDataItems.Length` times.  Do we *really* need to iterate this many times, especially when the following line skips all but one case?